rust/src/mlil/operation.rs

```diff
@@ -0,0 +1,1151 @@
+use binaryninjacore_sys::BNMediumLevelILInstruction;
+
+use core::marker::PhantomData;
+
+use crate::types::SSAVariable;
+
+use super::*;
+
+pub struct Operation<'func, A, M, F, O>
+where
+    A: 'func + Architecture,
+    M: FunctionMutability,
+    F: FunctionForm,
+    O: OperationArguments,
+{
+    pub(crate) function: &'func Function<A, M, F>,
+    pub(crate) op: BNMediumLevelILInstruction,
+    _args: PhantomData<O>,
+}
+
+impl<'func, A, M, F, O> Operation<'func, A, M, F, O>
+where
+    A: 'func + Architecture,
+    M: FunctionMutability,
+    F: FunctionForm,
+    O: OperationArguments,
+{
+    pub(crate) fn new(function: &'func Function<A, M, F>, op: BNMediumLevelILInstruction) -> Self {
+        Self {
+            function,
+            op,
+            _args: PhantomData,
+        }
+    }
+
+    pub fn address(&self) -> u64 {
+        self.op.address
+    }
+
+    pub fn size(&self) -> usize {
+        self.op.size
+    }
+
+    fn get_expr(&self, operand_index: usize) -> Expression<'func, A, M, F, ValueExpr> {
+        Expression::new(self.function, self.op.operands[operand_index] as usize)
+    }
+
+    fn get_int(&self, operand_index: usize) -> u64 {
+        // TODO check mask...
```

Comment:
The python API allow all values to be positive/negative, its unclear witch ones are signed/unsigned.